### PR TITLE
add a non-mutating probe_value method for fast paths

### PR DESCRIPTION
**Pull request by the8472**
The new API is needed for optimizations in https://github.com/rust-lang/rust/pull/108815

### CODE REVIEW

**Comment by the8472** [comment]
Also needs a version bump so we can use it. Can I include that in this PR or should that be done separately?

**Comment by nnethercote** [comment]
The most recent `ena` change was https://github.com/rust-lang/ena/pull/43, and @nikomatsakis then did a release off https://github.com/rust-lang/ena/pull/44.

File: src/unify/mod.rs [hunk 1]
```diff
@@ -447,6 +447,31 @@ impl<S: UnificationStoreMut> UnificationTable<S> {
...
+impl<S, K, V> UnificationTable<S>
+where
+    S: UnificationStoreBase<Key = K, Value = V>,
+    K: UnifyKey<Value = V>,
+    V: UnifyValue,
+{
+    /// Attempts to return the current value for the given key. If the key has
+    /// been union'd, this may give the current value from the current root or `None`.
+    ///
+    /// This is useful for fast-paths that want to avoid pointer-chasing.
```
Comment (nikomatsakis) [inline src/unify/mod.rs:459]:
```suggestion
    /// Obtains current value for key without any pointer chasing; may return `None` if key has been union'd.
```

How does this sound?

**Review by nikomatsakis — CHANGES_REQUESTED** [review]
I found the comment confusing. What do you think of this version, @the8472 ?

**Pull request merged**